source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -340,7 +345,13 @@ Http::FilterDataStatus AWSLambdaFilter::decodeData(Buffer::Instance &data,
     has_body_ = true;
   }
 
-  aws_authenticator_.updatePayloadHash(data);
+  // If we are not transforming the request, then update the payload hash according to the incoming data
+  // If we are transforming the request, then we will update the payload hash after the transformation
+  if (!isRequestTransformationNeeded()) {
```

Comment:
question should we just always move the data to request body and then call update with the whole body at finalization?

Reply:
see my comment below; we want to let enovy manage the request buffer. we can discuss